### ISSUE TITLE
Upgrade from 5.2.1 to 6.0.2 fails

### ISSUE THREAD

**Issue by TILvids** [YunoHost-Apps/peertube_ynh#416]
Here's the yunopaste. Any ideas?

https://paste.yunohost.org/raw/agabodumig

**Comment by isAAAc** [comment]
```
root@krashboyz:~# yunohost --version
yunohost: 
  repo: stable
  version: 11.2.9.1
yunohost-admin: 
  repo: stable
  version: 11.2.4
moulinette: 
  repo: stable
  version: 11.2
ssowat: 
  repo: stable
  version: 11.2
```

Same here 
https://paste.yunohost.org/raw/anovapozim

**Comment by isAAAc** [comment]
logs for `5.2.1~ynh1` to `6.0.2~ynh3` (KO too) 
https://paste.yunohost.org/raw/imejetaxij

**Comment by alexAubin** [comment]
`WARNING - ./upgrade: line 16: admin: unbound variable`

apparently it's an old install which didnt have the `admin` defined ...

you can define the admin with `yunohost app setting peertube admin -v some_admin_user` with `some_admin_user` being an existing yunohost user

maybe we should add PRE_UPGRADE warning for this specific version to explain this

**Comment by alexAubin** [comment]
Well uh yes that's curious .. root is hardcoded here : https://github.com/YunoHost-Apps/peertube_ynh/blob/master/doc/POST_UPGRADE.md?plain=1#L2

dunno if that's intentional or ...

**Comment by isAAAc** [comment]
@TILvids : does this command fix your issue too ?

**Comment by ashledombos** [comment]
I did not see this issue until now, I found a workaround it by adding 
`admin: someuser`
in `/etc/yunohost/apps/peertube/settings.yml`
It’s less elegant, but installation worked.

However, logins don’t work anymore (either an account, nor the original root account), but that’s another issue.

Note: from memory I had the same outpout as https://github.com/YunoHost-Apps/peertube_ynh/issues/416#issuecomment-1892148444, and also lost All Users access to Peertube, so had to add this access back by hand.
